tests/helpers_tests/test_condition.py

```diff
@@ -108,6 +112,206 @@ def test_evaluate__returns_null__when_condition_has_an_invalid_type_property(sel
+                             " {} and target version: {}".format(result,
+                                                                 user_version,
+                                                                 target_version
+                                                                 )
+            self.assertEqual(result, 0, custom_err_msg)
+
+    def test_compare_user_version_with_target_version_greater_than_0(self):
+        semver_greater_than_2_0_condition_list = [['Android', "2.0", 'custom_attribute', 'semver_gt']]
+        versions = [
+            ('2.0.0', '2.0.1'),
+            ('2.0', '3.0.1'),
+            ('2.0', '2.9.1'),
+            ('2.9.0', '2.9.1'),
+            ('2.1.2', '2.1.3-beta'),
+            ('2.1.2-beta', '2.1.2-release'),
+            ('2.1.3-beta', '2.1.3')
+        ]
+        for target_version, user_version in versions:
+            evaluator = condition_helper.CustomAttributeConditionEvaluator(
+                semver_greater_than_2_0_condition_list, {'Android': user_version}, self.mock_client_logger)
+            result = evaluator.compare_user_version_with_target_version(target_version, user_version)
+            custom_err_msg = "Got {} in result. Failed for user version:" \
+                             " {} and target version: {}".format(result,
+                                                                 user_version,
+                                                                 target_version)
+            self.assertEqual(result, 1, custom_err_msg)
+
+    def test_compare_user_version_with_target_version_less_than_0(self):
+        semver_greater_than_2_0_condition_list = [['Android', "2.0", 'custom_attribute', 'semver_gt']]
+        versions = [
+            ('2.0.1', '2.0.0'),
+            ('3.0', '2.0.1'),
+            ('2.3', '2.0.1'),
+            ('2.3.5', '2.3.1'),
+            ('2.9.8', '2.9'),
+            ('2.1.2-release', '2.1.2-beta'),
+            ('2.1.3', '2.1.3-beta')
+        ]
+        for target_version, user_version in versions:
+            evaluator = condition_helper.CustomAttributeConditionEvaluator(
+                semver_greater_than_2_0_condition_list, {'Android': user_version}, self.mock_client_logger)
+            result = evaluator.compare_user_version_with_target_version(target_version, user_version)
+            custom_err_msg = "Got {} in result. Failed for user version: {} " \
+                             "and target version: {}".format(result,
+                                                             user_version,
+                                                             target_version)
+            self.assertEquals(result, -1, custom_err_msg)
+
+    def test_compare_invalid_user_version_with(self):
+        semver_greater_than_2_0_condition_list = [['Android', "2.0", 'custom_attribute', 'semver_gt']]
+        versions = [
+            '-', '.', '..', '+', '+test', ' ', '2 .3. 0', '2.', '.2.2', '3.7.2.2', '3.x', ',', '+build-prerelease'
```

Comment:
Missing a test case that I mentioned in the last iteration. Something ike `2..2`

Reply:
will add that case.